internal/scorecard/plugins/plugin_runner.go

```diff
@@ -346,3 +348,47 @@ func RunInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig, lo
 	}
 	return output, nil
 }
+
+func ListInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig) (scapiv1alpha1.ScorecardOutput, error) {
+	var suites []schelpers.TestSuite
+
+	if pluginType == BasicOperator {
+		conf := BasicTestConfig{}
+		basicTests := NewBasicTestSuite(conf)
+		if schelpers.IsV1alpha2(config.Version) {
+			basicTests.ApplySelector(config.Selector)
+		}
+
```

Comment:
WDYT about add the logic in the new of object? Who knows more about the object than itself ;-)?

Reply:
ok

Reply:
Wouldn't that mean that we would have to pass version and selector to `NewBasicTestSuite`? If so, that seems like a worse API for that function.

Reply:
I added selector/version into the configs going into those New suite functions since they were already being passed in, but I'm pretty flexible here if you want me to change it back to the previous style let me know.

Reply:
Yeah, IMO, BasicTestSuite and OLMTestSuite don't need to know about the version and selector, so that's why I originally suggested to pull that stuff out.

I think part of the problem is that the setup code and this test suite stuff is in need of refactoring, and we're working with what we have for now.

In the long run, the way I envision this working is that we build the entire set of tests to list/run and then apply the selector once against the entire set.

I think we'll end up with something like this pseudocode:

```go
func (s Scorecard) Run(conf ScorecardConfig) (ScorecardOutput, error)
	tests := Tests{}
	for _, plugin := range conf.Plugins {
		pluginTests, err := plugin.GetTests()
		if err != nil {
			return err
		}
		tests = append(tests, pluginTests...)
	}
        if !isV1alpha1(conf.Version) {
                tests.ApplySelector(conf.Selector)
        }
	if conf.List {
		return tests.List()
	}
	return tests.Run()
}
```

Reply:
ok, changed it back to the previous....yeah, this stuff has refactor potential.

Reply:
Hi @joelanford,

> Wouldn't that mean that we would have to pass version and selector to NewBasicTestSuite?  

Not exactly. I was speaking about using objects/structures.

**NOTE**  RunInternalPlugin has  275 lines == high complexity == very low manutence ability  

My suggestion was doing something +/- like: 

```go
func (s Scorecard) Run(conf ScorecardConfig) (ScorecardOutput, error)
	tests := newTests(conf)
	err := tests.applyPlugins(conf)
         if err != nil {
	       return err
	}
        if conf.List {
		return tests.List()
	}
        return tests.Run()
}

func NewTests(ScorecardConfig conf) *Tests {
           tests := Tests{} 
	  if !isV1alpha1(conf.Version) {
                tests.ApplySelector(conf.Selector)
        }
}

func (t *Tests) applyPlugins(ScorecardConfig conf) (*Tests, error){
	for _, plugin := range conf.Plugins {
		pluginTests, err := plugin.GetTests()
		if err != nil {
			return err
		}
		t = append(t, pluginTests...)
	}
}
```
So, by following this approach, in the end, we should have the logic of `RunInternalPlugin` something as:

```go
switch pluginType {
	case BasicOperator:
              conf := BasicTestConfig{}
	     return NewBasicTestSuite(conf)
        case OLMIntegration:
             conf := OLMTestConfig{}
	     return  NewOLMTestSuite(conf)
....
```  
However, it is up to you folks, I am happy to move forward as you decide and then when we have time we can try to refact these parts which may make more sense. I really just would like to better clarify my previous suggestion.